### ISSUE TITLE
Accordion: Accessibility Issues

### ISSUE THREAD

**Issue by coliff** [trimble-oss/modus-web-components#2079]
The Accordion component is currently not accessible via screen readers. Changes needed:

- The clickable header should be a button
- The header button should have `aria-expanded="false"` when closed
- The expanded content should have an ID and the button should link to it with `aria-controls` - e.g. `aria-controls="expandedContentID"`
